### ISSUE TITLE
miner crash or shutdown

### ISSUE THREAD

**Issue by Az00Mine** [KlausT/ccminer#133]
Hello,
I got some problems with that miner, using neocrypt to mine vivo. Miner working for 15min to 20 hours (thats longest time) and aplication get error or just shuting down. 
Tested on versions 8.15, 8.17, 8.18. I got the same configuration on the second rig and everything works very well. 
Other miners works fine too
Any suggestions whats wrong? 
![error 1](https://user-images.githubusercontent.com/35180678/34645133-a2b762e4-f345-11e7-9458-a79f5314cdfa.jpg)

GPU: 5x MSI GTX 1070 Gaming Z
         1x MSI GTX 1070 Gaming X
CPU: Pentium G4400
MB; Asus PRIME Z270-A
RAM: 2x 4GB Kingston HyperFury X
PSU: 2x EVGA SUPERNOVA 750W G2
OS: Windows 10 Pro 64bit
Drivers: 384.76 nvidia
Afterburner Config:
power limit 65
temp limit 75
core clock 140
memory clock 500

EDIT:
Installed new nvidia drivers 388.71 and reduced core clock to 140, working now about 30min and maybe it will be ok

**Comment by chicri** [comment]
I have the same issue. Win10x64 with cuda9 8.17, 8.18

driver 388.59 - 10 x 1070ti's and 2 x 1070
power target 130W on all gpu via nvsmi no overclocking

I'm now trying cuda 8 version

EDIT: Not sure what else to do. Hopefully we have a new release coming to fix the crashing.
Currently installing fresh windows 1709 without updating it. Thats what I did with my other rig of 1070's. Will update if I'm able to resolve.

EDIT2: Fresh install worked so far. Installed windows 10 1709 without connecting to the internet then disabled windows update. Will see how long that will last.

EDIT3: trying to split 8gpu and 4gpu. set intensity to 15 for both bat file. above 15 it instantly crashed. It might be the intensity causing the issue but not sure yet. hope its still up and running when I wake up

**Comment by Az00Mine** [comment]
Update - It was mining since my last update here, crashed 5min ago... Maybe thats the network problem? I've static IP for miner. Should I unlock some ports in windows?

**Comment by KlausT** [comment]
Yeah, could be a firewall problem

**Comment by chicri** [comment]
I give up cant split my miner still crashed

**Comment by Az00Mine** [comment]
Try to configure your router/modem and turn on DMZ, I think it working for me for several hours now

**Comment by chicri** [comment]
for me I think its the cpu being over worked. I'm now running 10 gpus instead of 12 and seems to be ok @ 72% utilization. btw cpu is G4400. might need to change that to a quad core for a 12gpu build.

**Comment by lyolyalya** [comment]
altminer is not that stable as i want.sooo i'm just not mining on it and have no problems.

**Comment by chicri** [comment]
woke up and miner is dead 
 
EDIT: So I tried running 8 gpus and it seems to be working. My other rig is an 8 gpu as well no issues. The issue maybe running more than 8 gpus or the driver/OS cant handle it. Hope @KlausT can make a work around for it 👍 

EDIT: still crashing even with 8 gpus.

**Comment by KlausT** [comment]
What do you mean with "crashing" ?
Is there an error message?

**Comment by ultiman2** [comment]
i have the same kind of problem. after some time (serveral hours) the miner just exit without any error. im trying to change the intenisty and will report back if change anything. if you have any option or version that  create debug log i can try it

**Comment by chicri** [comment]
2 types of crashing
1. app hangs and closes no error. just a window close program from windows
2. complete power down of windows

resolution: used unix os simple mining instead of windows. have not seen a single app crash or system crash

loved this miner but couldnt do 24/7 operation

**Comment by KlausT** [comment]
> complete power down of windows

Well, in this case it's almost certainly either a driver issue or a hardware issue.
I can't remember my Windows shutting down while mining. Ever.

Closing without error message can happen, but I have no idea why it does that.

**Comment by chicri** [comment]
@KlausT yeah I think this is an issue with windows or the driver itself. Do you think the issue is cpu usage? I'm confused as hell whats up with it.

**Comment by ultiman2** [comment]
So after some more tests, i found that if i use -i 15 or 16 it will crash with no error message after some time. work ok if i dont choose -i. i mining on 2 x gtx 1060 6g with some oc. any way to some log and see if its a error in your program or if i just push the card too much?

**Comment by Az00Mine** [comment]
I unlocked some ports in firewall, miner works fine for more than 48h. Now found next little issue, it happens every 3-8min and have 2-3 duration.
  ![error cpu](https://user-images.githubusercontent.com/35180678/34746366-f6f71fde-f593-11e7-8ddf-51e3fad06548.jpg)

**Comment by KlausT** [comment]
@Az00Mine 
Is it always the same card?
Try reducing the overclock

**Comment by Az00Mine** [comment]
It happens only with GPU #3
Reduced Core Clock from 150 to 110 on this gpu and nothing happens, still error 
Despite the reducing core clock - hash rate didnt change (when reducing one gpu, if I reduce at all it goes lower)
Anyway what you think about this OC settings? I saw many people got power limit increased to 85-100%, I got 65% and my gpus consume about 140W each, so what if someone got 100% power limit? 150+W each gpu? I dont get it

**Comment by KlausT** [comment]
Try reducing the memory clock for GPU #3

I don't have much experience with overclocking, but I can say that each card model is behaving differently, depending on if it's factory overclocked or not, and how good the cooling is.

**Comment by Az00Mine** [comment]
"GPU Does not validate on CPU" error fixed by reinstalling drivers to ver. 384.76
"Stratum connection intrerupted" error fixed by configuring some network stuff in my router (!@#$ $%# orange)
Still got the problem with miner - randomly closing the miner without any errors. It works between 1-30h...  Tested versions from 8.15 to 8.18. Used many configs with OC, even lowest settings on gpus.
Before ccminer I was mining ZEC and BTG with EBWF for 5-6 month without any problems ;s

Anyway REALLY appreciate it KlausT for responding and trying to help with that issue!
Tried ccminer from tpruvot but I can get max 5600kh/s (at yours 7000kh/s) I dont know whats wrong.

And do you get any idea for setting option - when miner turns off -> it will start again automatically?
Cheers! and sorry for bad english ;)

**Comment by ultiman2** [comment]
for me the ccminer only close by itself after some time if i change the -i
to 15 or 16, if i dont choose any value (i dont know the default one) it
run for a very long time without any problem.  Klaus can you tell me the
default intensity or add a fonction to show it

thanks again for your great work

**Comment by KlausT** [comment]
@Az00Mine When you are using a bat file to run the miner you could make a loop like this:

```
@echo off
:start
ccminer.exe -your -options -here
rem here we can wait some seconds until we start again (optional)
timeout 10
goto start
```

@ultiman2 I will see if I can show the default intensity

**Comment by Az00Mine** [comment]
@KlausT  I made a loop that you suggest, it works, miner starts over again. By using this method I found what error is showing up just before miner quit 
![bez tytulu](https://user-images.githubusercontent.com/35180678/34800697-40e1ef62-f665-11e7-864d-f5e12e57aa3a.png)

**Comment by dedozeno** [comment]
too much core oc or check the riser or power cable on gpu 3 , i had problem like this on one of mine ,my case was bad cable , but times before that it was same k4 error and it was core oc

**Comment by david-987** [comment]
I have the same/similar issue as the original post, ccminer exe crashes when it loses stratum connection. Happens on bsod.pw pool every 24 hours or so, will attach screenshot next time it happens.
Happens on all 3 of my rigs, 1070, 1070 ti and 1080ti's

**Comment by Az00Mine** [comment]
@dedozeno Its a problem with memory clock, I reduced it from 600 to 150 and now works fine without any hashrate decrease
@dunnies Propably you have configured -p 0 or -p 2 at the bat file, right? I deleted it
I unclocked some ports and miner in firewall - fixed problem

**Comment by chicri** [comment]
I'm running 10 gpus on my rig. solved problem by setting virtual memory to 90,000 MB. I'm not sure how to calculate the correct virtual memory though.
